### PR TITLE
:sparkles: PRTL-1903. Adds secondary table for multi primary sites

### PR DESCRIPTION
**Pull request by anncatton**

### CODE REVIEW

File: src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* @flow */
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js]: I think this file should be renamed. it's more like a disease list right?

File: src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.relay.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      ['location'],
+      ({ location: { search }, defaultFilters = null }) => {
+        const q = parse(search);
+        const filters = parseFilterParam(q.filters, defaultFilters);
+        return {
+          filters,
+        };
+      },
+    ),
+    // withPropsOnChange(
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.relay.js]: 🚫

File: src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.relay.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          },
+        };
+      },
+    ),
+  )((props: Object) => {
+    return (
+      <Query
+        name="CasesByPrimarySite"
+        parentProps={props}
+        // minHeight={249}
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.relay.js]: don't need this?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            </Td>
+          </Tr>
+        )}
+      </tbody>
+    }
+  />
+);
+
+export default compose(
+  // setDisplayName('ProjectPrimarySitesTablePresentation'),
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: why commented?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  branch(
+    ({ viewer }) =>
+      !viewer.projects.hits.edges ||
+      viewer.projects.hits.edges[0].node.primary_site.length <= 1,
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: it should render nothing if only 1?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+  connect(state => ({
+    tableColumns: state.tableColumns.projectPrimarySites.ids,
+  })),
+  mapProps(props => ({
+    ...props,
+    hits: props.viewer.projects.hits,
+  })),
+)(
+  ({
+    viewer: { projects: { hits: { edges } } },
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: you map `hits` right above here but then don't use it?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    tableHeader = 'Primary Sites',
+    projectId,
+    searchValue,
+    setSearchValue,
+    theme,
+    loading,
+  }) => {
+    const project = edges[0].node;
+    const tableInfo = tableModels[entityType]
+      // .sort((a, b) => tableColumns.indexOf(a.id) - tableColumns.indexOf(b.id))
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: don't need?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+      .filter(x => tableColumns.includes(x.id));
+
+    const primarySiteData = project.primary_site
+      .filter(site => site.toLowerCase().includes(searchValue.toLowerCase()))
+      .sort();
+
+    const paginationPrefix = 'ps';
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: can you make this `primarySites`

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+              field: 'cases.primary_site',
+              value: props.primarySite,
+            },
+          ]),
+        }}
+      >
+        {props.repository.cases.aggregations.disease_type.buckets.length
+          ? props.repository.cases.aggregations.disease_type.buckets[0]
+              .doc_count
+          : '0 --'}
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js]: should this just be `--` or 0 depending on if its missing or 0 length

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+  },
+  {
+    name: 'Data Categories',
+    id: 'data_categories',
+    sortable: false,
+    downloadable: false,
+    th: ({ primarySite, projectId }) => (
+      <Th rowSpan="2">
+        Available Cases per Data Category
+        {/* <span
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js]: don't need?

File: src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js (hunk 1 above)
Comment (anncatton) [reply src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js]: I'm wondering if I need it at all? I'm just using the relay query in the table model, but not the component.
And yeah, I can rename it, maybe DiseaseListByPrimarySite ?

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js (hunk 5 above)
Comment (anncatton) [reply src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: Yeah. The table is only supposed to show if there are multiple primary sites.

Comment (alex-wilmer) [reply src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.js]: can that be changed to `< 2` then

File: src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js (hunk 1 above)
Comment (alex-wilmer) [reply src/packages/@ncigdc/modern_components/CasesByPrimarySite/CasesByPrimarySite.js]: if you're not using it, then yea just remove it

File: src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js (hunk 9 above)
Comment (anncatton) [reply src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js]: I checked, this was actually supposed to be just '0' if no files present.

File: src/packages/@ncigdc/modern_components/environment.js [hunk 11]
```diff
@@ -32,7 +59,9 @@ function fetchQuery(operation, variables, cacheConfig) {
   }).then(response =>
     response.json().then(json => {
       if (response.status === 200) {
+        // if the response is ok, and the result to the simpleCache and delete it from the pendingCache
```
Comment (alex-wilmer) [inline src/packages/@ncigdc/modern_components/environment.js]: add the result?

Comment (anncatton) [reply src/packages/@ncigdc/modern_components/environment.js]: Darn - yes, add.

**Pull request merged**